### PR TITLE
Fix some issues with posix_sockets.h

### PR DESCRIPTION
**Pull request by mpfj**

### CODE REVIEW

**Comment by LiamBindle** [comment]
Thanks for the PR @mpfj! Could you expand on what you fixed?

File: examples/templates/posix_sockets.h [hunk 1]
```diff
@@ -34,14 +34,20 @@ int open_nb_socket(const char* addr, const char* port) {
...
     }  
 
     /* free servinfo */
     freeaddrinfo(servinfo);
 
+	/* did connect fail? */
+	if (!p) {
+		close(sockfd);
+		sockfd = -1;
+	}
```
Comment (LiamBindle) [inline examples/templates/posix_sockets.h:49]: This indent has 2 extra spaced. Do you mind fixing this?

Comment (mpfj) [reply examples/templates/posix_sockets.h:49]: Not sure what you mean.  Is it because I used tab indents instead of spaces ?

**Comment by mpfj** [comment]
In 50e4933 I have trapped for when the previous for() loop fails to successfully connect, leaving p == NULL.
If this happens, the socket is left open and returned even though no connection has been made.
I would have thought if the connect() fails, the socket should be closed and return -1 (to indicate there was a fault).

In 1c96e50 the for() loop goes through the available addrinfo items, using p to point to the current item.
The call to socket() correctly uses "p->..." but the connect() uses "servinfo->..." which always points to the first addrinfo item, not the current one pointed to by p.

**Comment by LiamBindle** [comment]
Thanks for the contribution @mpfj! Hope to see more PRs!

**Pull request merged**